lib/AST/Type.cpp

```diff
@@ -1246,13 +1254,14 @@ CanType TypeBase::computeCanonicalType() {
     SmallVector<AnyFunctionType::Param, 8> canParams;
     getCanonicalParams(funcTy, genericSig, canParams);
     auto resultTy = funcTy->getResult()->getCanonicalType(genericSig);
+    auto throwsTy = funcTy->getThrowsType()->getCanonicalType(genericSig);
```

Comment:
this fails always in tests

Reply:
You cannot get the canonical type from a null `Type`. Try:
```
CanType throwsTy;
if (auto funcThrowsTy = funcTy->getThrowsType())
  throwsTy = funcThrowsTy->getCanonicalType(genericSig);
```

Reply:
What happens with all these traces, is Never not usable to be injected as a default throwsType, even if the function doesn't throw:
```
swift-frontend           0x00000001039995ac swift::DeclContext::getAsDecl() + 28
6  swift-frontend           0x00000001039ebbd5 swift::DeclContext::getAsDecl() const + 21
7  swift-frontend           0x0000000103c751d5 swift::DeclContext::isModuleContext() const + 21
8  swift-frontend           0x0000000103d8d3dd swift::DeclContext::getParentModule() const + 29
9  swift-frontend           0x0000000103d8d3a5 swift::DeclContext::getASTContext() const + 21
10 swift-frontend           0x000000010397bfae swift::Decl::getASTContext() const + 62
11 swift-frontend           0x0000000103c55f34 computeNominalType(swift::NominalTypeDecl*, DeclTypeKind) + 36
12 swift-frontend           0x0000000103c5576d swift::NominalTypeDecl::getDeclaredInterfaceType() const + 93
13 swift-frontend           0x00000001039755bf swift::ASTContext::getNeverType() const + 63
```